pkg/querier/queryrange/split_by_interval.go

```diff
@@ -166,22 +166,41 @@ func (h *splitByInterval) Do(ctx context.Context, r queryrange.Request) (queryra
 		return h.next.Do(ctx, r)
 	}
 
-	intervals := splitByTime(lokiRequest, interval)
-	h.metrics.splits.Observe(float64(len(intervals)))
+	var intervals []queryrange.Request
+	var limit int64
+	switch req := r.(type) {
+	case *LokiRequest:
```

Comment:
I don't think you need all this duplicate logic:
```go
		intervals = splitByTime(req, interval)
		h.metrics.splits.Observe(float64(len(intervals)))
		// no interval should not be processed by the frontend.
		if len(intervals) == 0 {
			return h.next.Do(ctx, r)
		}
```

Instead I think you can perform just one type switch, checking if it's a series request and then setting the limit to 0 as all the other logic is identical